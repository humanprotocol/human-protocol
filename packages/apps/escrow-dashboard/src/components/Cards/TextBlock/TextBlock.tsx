import { Box, Typography } from '@mui/material';
import numeral from 'numeral';
import { FC, ReactElement } from 'react';
import Skeleton, { SkeletonTheme } from 'react-loading-skeleton';

import { Container } from '../Container';
import { TooltipIcon } from 'src/components/TooltipIcon';

type TextBlockProps = {
  title: string | ReactElement;
  value?: number | string;
  component?: any;
  format?: string;
  changes?: number;
  tooltipTitle?: string;
};

export const TextBlock: FC<TextBlockProps> = ({
  title,
  value,
  component: ValueComponent,
  format = '0,0',
  changes,
  tooltipTitle,
}) => {
  return (
    <Container>
      <Typography variant="body2" color="primary" fontWeight={600}>
        {title}
      </Typography>
      {value === undefined ? (
        <Box
          sx={{ marginTop: { xs: '4px', md: 2 }, height: { xs: 50, md: 72 } }}
        >
          <SkeletonTheme
            baseColor="rgba(0, 0, 0, 0.1)"
            highlightColor="rgba(0, 0, 0, 0.18)"
          >
            <Skeleton count={1} width="100%" height="100%" />
          </SkeletonTheme>
        </Box>
      ) : (
        <Box display="flex" alignItems="baseline" overflow="hidden">
          {ValueComponent ? (
            <ValueComponent value={value} />
          ) : (
            <Typography
              variant="h2"
              color="primary"
              lineHeight={1.2}
              marginTop={{ xs: '4px', md: 2 }}
            >
              {Number.isNaN(Number(value))
                ? value
                : numeral(value).format(format)}
            </Typography>
          )}

          {changes === undefined ? (
            <></>
          ) : changes >= 0 ? (
            <Typography variant="caption" color="success.light" ml="4px">
              (+{Math.abs(changes).toFixed(2)}%)
            </Typography>
          ) : (
            <Typography variant="caption" color="error.light" ml="4px">
              (-{Math.abs(changes).toFixed(2)}%)
            </Typography>
          )}
        </Box>
      )}
      {tooltipTitle && <TooltipIcon position="topRight" title={tooltipTitle} />}
    </Container>
  );
};
