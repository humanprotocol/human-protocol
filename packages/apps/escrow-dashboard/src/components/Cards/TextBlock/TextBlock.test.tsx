import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';

import { TextBlock } from './TextBlock';

const mock = {
  value: 'Value',
  title: 'Title',
};

describe('when rendered CardBarChart component', () => {
  it('should render passed prop `value`', () => {
    render(<TextBlock value={mock.value} title={mock.title} />);
    expect(screen.findByLabelText(mock.value)).toBeTruthy();
  });

  it('should render passed prop `title`', () => {
    render(<TextBlock value={mock.value} title={mock.title} />);
    expect(screen.findByLabelText(mock.title)).toBeTruthy();
  });
});

it('TextBlock component renders correctly, corresponds to the snapshot', () => {
  const tree = create(
    <TextBlock value={mock.value} title={mock.title} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
