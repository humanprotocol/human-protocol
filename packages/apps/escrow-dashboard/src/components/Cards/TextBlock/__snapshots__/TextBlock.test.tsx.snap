// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextBlock component renders correctly, corresponds to the snapshot 1`] = `
<div
  className="MuiBox-root css-19n1ffs"
>
  <p
    className="MuiTypography-root MuiTypography-body2 css-ngc1bc-MuiTypography-root"
  >
    Title
  </p>
  <div
    className="MuiBox-root css-h43ml6"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2 css-14wx86a-MuiTypography-root"
    >
      Value
    </h2>
  </div>
</div>
`;
